sdks/python/apache_beam/runners/worker/statesampler.pyx

```diff
@@ -252,12 +260,14 @@ cdef class ScopedState(object):
     self.old_state_index = self.sampler.current_state_index
     pythread.PyThread_acquire_lock(self.sampler.lock, pythread.WAIT_LOCK)
     self.sampler.current_state_index = self.state_index
+    self.sampler.time_since_transition = 0
```

Comment:
<!--thread_id:cc_143602537_t; commit:5275ff1d3cf172be6722b6a37e6b416e58bcd4f4; resolved:1-->
<!--section:context-quote-->
> **charlesccychen** wrote:
> Instead of doing this reset to 0 here, can you do this in the `StateSampler.run()`?  This would avoid increasing the per-transition overhead.  For example, right before line 164, you could check if `self.state_transition_count` has changed from its previous value (which you can store as a local variable in that method)--if it has, then reset `time_since_transition` to 0.

<!--section:body-->
Done.